src/pybind/mgr/prometheus/module.py

```diff
@@ -110,40 +110,30 @@ def health_status_to_number(status):
 class Metrics(object):
     def __init__(self):
         self.metrics = self._setup_static_metrics()
-        self.pending = {}
 
-    def set(self, key, value, labels=('',)):
+    def clear(self):
```

Comment:
That would mean that we would have to setup static metrics (and all the other metrics) on each scrape. I did it this way to avoid the overhead of creating all the Metric instances on each scrape.